src/export_data.py

```diff
@@ -0,0 +1,122 @@
+                        default=10)
+
+    args = parser.parse_args()
+
+    params = json.load(open(args.params_path, 'r'))
+    BUCKET = args.bucket
+    DIRECTORY = args.export_folder
+    num_samples = args.num_samples
+    num_shards = args.num_shards
+
+    print("Loading", (args.params_path))
+    params_path = args.params_path
+    params = json.load(open(params_path, 'r'))
+    PATCH_BANDS = []
+    PATCH_BANDS += params['bands']['multispectral']
+    PATCH_BANDS += params['bands']['tropomi']
+    PATCH_BANDS += params['bands']['road'] + params['bands']['dsm']
+    STACKED_WIND_BANDS = ["%i_%s" % (i, b) for b in params['bands']['wind']
+                          for i in range(12)]
+    PATCH_BANDS += STACKED_WIND_BANDS
+    BANDS = PATCH_BANDS + ["HOD", "DOW", "DOM", "MOY", "latitude", "longitude",
+                           "valid"]
+    SCALE = params['scale']
+    KERNEL_RADIUS = params['kernel_radius']
+    # Initialize EE
+    ee.Authenticate()
+    ee.Initialize()
+    # Creating diffrent kernels
+    kernel = ee.Kernel.square(KERNEL_RADIUS)
+    # vertical and horizontal kernel used in calculating valid pixels
+    vertical_kernel = ee.Kernel.rectangle(xRadius=1,
+                                          yRadius=KERNEL_RADIUS,
+                                          units='pixels')
+    horizontal_kernel = ee.Kernel.rectangle(xRadius=KERNEL_RADIUS,
+                                            yRadius=1,
+                                            units='pixels')
+    # Initializing smaple exporter and TaskManager
+    task_manager = TaskManager(verbose=True)
+    sampler = SampleExporter(task_manager, num_samples, num_shards,
+                             kernel, SCALE, BUCKET, DIRECTORY)
+    # Initializing imagery classes
+    multispectral = MultiSpectralImagery(params['collections']['multispectral'],
+                                         start_date="2018-01-01",
+                                         end_date="2019-12-31",
+                                         bands=params['bands']['multispectral'],
+                                         scale=SCALE)
+    tropomi = TropomiImagery(params['collections']['tropomi'],
+                             bands=params['bands']['tropomi'])
+    # Wind imagery adds wind bands from the previous 12 hours
+    wind = WindImagery(params['collections']['wind'],
+                       bands=params['bands']['wind'])
+    # DSM Imagery returns the most recent data prior to the tropomi date
+    dsm = DSMImagery(params['collections']['dsm'], SCALE,
+                     bands=params['bands']['dsm'])
+    # road is an image, returns clipped image cropped to multispectral region
+    road = RoadImagery(params['collections']['road'],
+                       bands=params['bands']['road'])
+
+    print("Stacking bands for %i multispectral images" % len(multispectral))
+
+    # First 2 images already exported
+    for j in range(2, len(multispectral)):
```

Comment:
There are two for loops going on here, can we add a comment explaining what is happening in these two for-loops?